packages/astro/src/core/build/static-build.ts

```diff
@@ -253,15 +253,14 @@ async function copyFiles(fromFolder: URL, toFolder: URL) {
 		cwd: fileURLToPath(fromFolder),
 	});
 
-	// Make the directory
-	await fs.promises.mkdir(toFolder, { recursive: true });
-
 	await Promise.all(
 		files.map(async (filename) => {
 			const from = new URL(filename, fromFolder);
 			const to = new URL(filename, toFolder);
-			return fs.promises.copyFile(from, to);
-		})
+			const toSubPath = new URL(npath.dirname(filename), toFolder)
```

Comment:
So I'm not sure why URL() was used. Semantically confusion but perhaps used as quick platform independent way to to join paths? Anyway I followed the style.

Result is a tad inefficient as for every file will travel the entire path from file system root creating required folders.

Reply:
Correct, we use `file://` `URL`s internally as a platform independent way to represent any path, then convert them back to paths as needed using `fileURLToPath(urlValue)`.

Reply:
My gut reaction is that `new URL('../', filename)` would be better than `npath.dirname(filename)` here, but I'm not positive without taking this branch for a spin locally.

Reply:
hah! We were all fooled it's actually the to file's folder, ie './' :)